### PR TITLE
*: decouple operator handler and server

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5839

### What is changed and how does it work?

1. move operator handler from server to pkg/operator/handler
2. move error from server to pkg/errs
3. add a interface for pd server and scheduling server

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -145,7 +145,7 @@ require (
 	github.com/pelletier/go-toml/v2 v2.0.1 // indirect
 	github.com/petermattis/goid v0.0.0-20211229010228-4d14c490ee36 // indirect
 	github.com/pingcap/tipb v0.0.0-20220718022156-3e2483c20a9e // indirect
-	github.com/pkg/errors v0.9.1 // indirect
+	github.com/pkg/errors v0.9.1
```
Comment (rleungx) [inline go.mod:148]: Don't import it

Comment (lhy1024) [reply go.mod:148]: why

Comment (rleungx) [reply go.mod:148]: We use pingcap/errors

Comment (lhy1024) [reply go.mod:148]: Maybe because move operator error from store to pkg/errs?

File: pkg/errs/error.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errs
+
+import "github.com/pingcap/errors"
+
+var (
+	// ErrOperatorNotFound is error info for operator not found.
+	ErrOperatorNotFound = errors.New("operator not found")
```
Comment (rleungx) [inline pkg/errs/error.go:21]: What is the difference between errors.go and errno.go?

Comment (lhy1024) [reply pkg/errs/error.go:21]: This pr just moves the code to decouple without changing it, do you think the two should be merged?

File: server/handler.go [hunk 3]
```diff
@@ -50,44 +45,41 @@ import (
...
+	*Server
+}
+
+func (s *server) GetCoordinator() *schedule.Coordinator {
+	return s.GetRaftCluster().GetCoordinator()
+}
+
+func (s *server) GetCluster() sche.SharedCluster {
+	return s.GetRaftCluster()
+}
```
Comment (rleungx) [inline server/handler.go:61]: Why do we need it?

File: pkg/errs/error.go (hunk 2 above)
Comment (rleungx) [reply pkg/errs/error.go:21]: If you want to move it to pkg/errs, IMO, we need to.

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**